extensions/cdc-core/src/main/java/com/hazelcast/jet/cdc/ChangeEventElement.java

```diff
@@ -0,0 +1,189 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.cdc;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.hazelcast.jet.annotation.EvolvingApi;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Arbitrary part of a {@link ChangeEvent}, as big as the whole body or
+ * as small as a single patch expression, based on a complete JSON
+ * expression. Contains various methods for retrieving component values
+ * or for mapping itself to data objects.
+ *
+ * @since 4.1
+ */
+@EvolvingApi
+public interface ChangeEventElement {
```

Comment:
how is ChangeEventElement different than JsonNode? I don't see anything specific to a ChangeEvent in this interface

Reply:
Made more sense when we didn't yet exclude MongoDB. There it had nothing to do with JsonNode. But even now, do we really want to have `JsonNode` in our API? Even if we do the same thing?

Reply:
I don't see an issue, it's well documented and has more capabilities? I believe we are going to be shading Jackson as part of our public API?

Reply:
This is not exactly how I ended up, `JsonNode` is gone too now, since moving to Jackson jr, but `ChangeEventElement` is greatly simplified now, we no longer implement parsing shenanigans, everything is in Jackson code. Take a look if you can.